### ISSUE TITLE
Real time monitoring page 'find logs' 500 server error

### ISSUE THREAD

**Issue by simonmeadows** [Vadims06/topolograph#52]
I have topolograph-docker running and ospfwatcher running.
I can see changes logged in my ELK stack to I know ospf watcher is working.
I can see changes creating entries in my mongo database.
When I click on 'Find Logs' on the real-time monitoring page I get a 500 internal server error response from flask.
The output of the flask container shows the following:
```shell
[2024-11-04 18:10:10 +0000] [7] [ERROR] Error handling request /upload-monitoring-stat
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/sync.py", line 134, in handle
    self.handle_request(listener, req, client, addr)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/sync.py", line 175, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/usr/local/lib/python3.9/site-packages/connexion/apps/abstract.py", line 249, in __call__
    return self.app(environ, start_response)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "routes.pyx", line 2001, in routes.upload_monitoring_stat
  File "routes.pyx", line 1973, in routes.upload_monitoring_stat._do_prep_link_timeframe
  File "Helpers.pyx", line 176, in Helpers._get_visjs_edge_id
  File "/usr/local/lib/python3.9/ipaddress.py", line 53, in ip_address
    raise ValueError(f'{address!r} does not appear to be an IPv4 or IPv6 address')
ValueError: '' does not appear to be an IPv4 or IPv6 address
```

the events in the mongo database look like this:
```json
{
	"_id": ObjectId("6729058f411c4525cf000001"),
	"host": "002077017f67",
	"area_num": "0.0.0.0",
	"watcher_time_iso8601": ISODate("2024-11-04T17:34:06Z"),
	"event_object": "100.64.1.0/24",
	"subnet_type": "internal",
	"graph_time": "04Nov2024_17h33m11s_46_hosts",
	"event_name": "network",
	"event_detected_by": "10.65.54.190",
	"asn": "123",
	"@timestamp": ISODate("2024-11-04T17:34:06.754Z"),
	"watcher_name": "ospfwatch",
	"path": "/home/watcher/watcher/logs/watcher2-gre1-ospf.ospf.log",
	"event_status": "changed",
	"old_cost": "50000",
	"int_ext_subtype": "0",
	"new_cost": "50001",
	"watcher_time": "2024-11-04T17:34:06Z"
}
```
Any suggestions as to what I need to do?

**Comment by Vadims06** [comment]
Hi @simonmeadows , thank you for submitting this issue.
oh yes, it happens because of recent OSPF/IS-IS Watcher log format change. I adjusted log import on Monitoring page accordingly. I released `vadims06/topolograph:2.40_dev`, please try this version. If it works - I will promote it as latest.

**Comment by simonmeadows** [comment]
Thanks @Vadims06 for the quick response.
It did work for the first try but after adding a cost change, I'm back to 500s again, this time it looks like the creation of a 'COST_CHANGE' object is the problem

```
[2024-11-05 08:29:12 +0000] [9] [ERROR] Error handling request /upload-monitoring-stat
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/sync.py", line 134, in handle
    self.handle_request(listener, req, client, addr)
  File "/usr/local/lib/python3.9/site-packages/gunicorn/workers/sync.py", line 175, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/usr/local/lib/python3.9/site-packages/connexion/apps/abstract.py", line 249, in __call__
    return self.app(environ, start_response)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "routes.pyx", line 2012, in routes.upload_monitoring_stat
  File "Helpers.pyx", line 2470, in Helpers.MonitoringDashboard.do_prepare_topology_by_requested_event_start_time
  File "Helpers.pyx", line 2609, in Helpers.MonitoringDashboard.get_all_events_of_graph
  File "Helpers.pyx", line 2577, in Helpers.MonitoringDashboard._get_logs_in_time_range
  File "Helpers.pyx", line 2315, in Helpers.COST_CHANGE.__init__
TypeError: __init__() takes at least 12 positional arguments (8 given)
```

**Comment by Vadims06** [comment]
Right, it required more changes than I originally thought :) I published `vadims06/topolograph:2.40` and did pre-release https://github.com/Vadims06/topolograph/releases/tag/v2.40. Please let me know if it works as expected.

**Comment by simonmeadows** [comment]
Thanks again @Vadims06 it works perfectly now.
I really appreciate the effort you have put into making such a useful tool.
I am re-costing an entire OSPF area next weekend and this will be invaluable in tracking all the changes.
